### PR TITLE
Properties & Incompatible Types in Assignment

### PR DESCRIPTION
**Pull request by nnadeau**
Documentation for https://github.com/python/mypy/issues/4167

### CODE REVIEW

File: docs/source/common_issues.rst [hunk 1]
```diff
@@ -472,3 +472,41 @@ Here's the above example modified to use ``MYPY``:
 
    def listify(arg: 'bar.BarClass') -> 'List[bar.BarClass]':
        return [arg]
+
+Properties & Incompatible Types in Assignment
+---------------------------------------------
+
+Consider the code below:
```
Comment (gvanrossum) [inline docs/source/common_issues.rst]: I think the whole thing is a bit too verbose. You should start with something like "The type of a property is taken from the getter method." Then the example should be more compact (compare some of the other sections in this file). The method bodies could just be `...` and the "main" section shouldn't have `if __name__...` .

File: docs/source/common_issues.rst [hunk 2]
```diff
@@ -472,3 +472,41 @@ Here's the above example modified to use ``MYPY``:
...
+    
+        @foo.setter
+        def foo(self, value: Union[int, float]) -> None:
+            self._foo = int(value)
+    
+    
+    if __name__ == '__main__':
+        a = A()
+        a.foo = 123
+        a.foo = 456.789
```
Comment (gvanrossum) [inline docs/source/common_issues.rst]: Add a comment here like "# error here".

File: docs/source/common_issues.rst [hunk 3]
```diff
@@ -472,3 +472,41 @@ Here's the above example modified to use ``MYPY``:
...
+    
+
+Running Mypy against the above code will generate the following error at ``a.foo = 456.789``: 
+
+.. code-block:: none
+    
+    error: Incompatible types in assignment (expression has type "float", variable has type "int")
+
+While overloading class property setters in Python is valid, Mypy infers the type from the initial ``@property`` definition (e.g., ``int``) as the final type for the attribute, regardless of what the ``@property.setter`` declares (e.g., ``Union[int, float]``). This error may be silenced with `# type: ignore  <http://mypy.readthedocs.io/en/latest/common_issues.html#spurious-errors-and-locally-silencing-the-checker>`_
```
Comment (gvanrossum) [inline docs/source/common_issues.rst]: Please break this line so each line is under 80 characters, and remove the remark about silencing it with `# type: ignore` -- by the time a user gets here they should know they can silence anything that way.

Also, "mypy" is not capitalized except at the start of a sentence.

**Comment by ilevkivskyi** [comment]
I thought the section "Common issues" is not about _known issues_ (i.e. bugs), but for things that are either correct but unintuitive for some developers (like invariants of ``List[T]``), or are some fundamental limitations (like import cycles). Most sections have motivation and/or workarounds.

Do we really need this? It looks just like a duplicate of the issue.

**Comment by nnadeau** [comment]
@ilevkivskyi , https://github.com/python/mypy/issues/4167 is currently being considered as a low-priority feature request and not a bug. IMHO, I think this documentation for the current state of mypy is useful for others who may stumble upon issues with properties, as it may change their approach to class architecture.

**Comment by gvanrossum** [comment]
Hm, I agree with Ivan that the bar for inclusion of something in the common_issues page should be higher than "low priority feature request" (else half of the issues in the tracker would deserve inclusion here). So I'll close the PR now without merging. If we see an uptick in reports of users who are confused by this we can reconsider.

**Pull request closed without merging**